### ISSUE TITLE
Problem in stride calculation when plot rgl object in Shiny

### ISSUE THREAD

**Issue by toduckhanh** [dmurdoch/rgl#173]
I use R 4.1.2 (window 10) and rgl (v 0.108.10). When running demo(simpleShinyRgl), if I check box "Rescale", one message appeared as: "Problem in stride calculation", then if I press "Ok", the 3D plot presented but without the axes (xyz) or the axes were in wrong place. On the other hand, with my shiny app, the 3D rgl plot is fine at the first time of plotting, but if I re-plot, the mentioned message will appear and the xyz axes will not displayed. I have no idea why. Could you please check this issue?

**Comment by dmurdoch** [comment]
Thanks for the report.  That's an unreleased version, so I'll keep it in mind for testing before the next release.  In the meantime, could you please check if you have the same issue after installing from CRAN?

**Comment by toduckhanh** [comment]
I had this issue with current version in CRAN (v 0.108.3). Then, I tried to insall the newest verison of rgl relasted in github. But, the issue is still there.

**Comment by dmurdoch** [comment]
Thanks, I'll take a look.

**Comment by dmurdoch** [comment]
I've just committed a fix to the issue173 branch.  It fixes this bug, but exposes a related one, so things aren't working yet.  

Now you don't get the "Problem" message, but the axes are all messed up.  I probably won't get to the second one today, so I wanted to record what's likely happening:

Running in Shiny is unusual in that scenes get initialized again when you make a change.  The stride issue happened because the second initialization was working with data that had been changed in later steps.  This doesn't matter outside of Shiny, because initialization only happens once.  I expect the current problem is something similar, but I haven't tracked down exactly what is being messed up.
